location/lbs/activity/audioset/dataprocessing/audio_processing.py

```diff
@@ -0,0 +1,577 @@
+
+        [[1,2,3,4],[1,3,5,2]] or [1,5,3,2,6,9]
+    """
+    if len(alist) == 1 and isinstance(alist, list):
+        return delete_outer_list(alist[0])
+    return alist
+
+
+def print_flags():
+    """Prints out the flags inputted via command line arguments.
+
+    Prints a filename, a list of labels, the redo boolean, a source
+    directory, a destination directory, the scaled boolean, and a list of
+    features, based on the command line input or their default values.
+    """
+    print(FLAGS.filename)
+    print(FLAGS.labels)
+    print(FLAGS.redo)
+    print(FLAGS.src_dir)
+    print(FLAGS.dest_dir)
+    print(FLAGS.scaled)
+    print(FLAGS.features)
+
+
+def output_df(src_dir, dest_dir, filename, labels, features_to_extract,
+              scaled=True, redo=False):
+    """Creates dataframe object from inputted csv files, features, and labels.
+
+        Parses through a csv file and extracts all the metadata from it. Then
+        iterates through the list of metadata confirming that each video
+        associated with every video_id is downloaded and attempt to download if
+        the video is missing. The specified features are then extracted from the
+        downloaded videos, and the features are then put into a dataframe with
+        a label of 1 or 0 depending on if the video's label is in the list of
+        labels passed in.
+
+        Args:
+            src_dir: Path to where all the input files are expected to be.
+            dest_dir: Path to where all the output files will be stored.
+            filename: Name of the audioset csv file not including '.csv'
+            labels: A list of labels to treat as positive examples.
+            features_to_extract: A list of features to extract.
+            scaled: A boolean on whether to average the extracted features.
+            redo: A boolean on whether to re-download all the videos.
+
+        Returns:
+            A pandas dataframe object with the following format:
+
+            LABEL   FEATURE1   FEATURE2   ...
+            0       [1, 3, 5]  [2, 4, 5]  ...
+            1       [1, 1, 5]  [4, 4, 5]  ...
+            0       [1, 3, 5]  [1, 4, 5]  ...
+        """
+    begin_time = datetime.datetime.now()
+    path = join(dest_dir, filename + '_features.csv')
+    if isfile(path) and not redo:
+        datasetdf = pd.read_csv(path)
+        return datasetdf
+    dataset = []
+    label_dict = get_label_dict(src_dir)  # plain English to audioset syntax
+    labels_set = label_list_to_set(labels, label_dict)  # user specified labels
+    audio_dict = parse_metadata(src_dir, filename)  # video_id to AudioSetEntry
+    count = 0
+    vid_count = 0
+    download_from_list(dest_dir, audio_dict, redo)
+    download_time = datetime.datetime.now() - begin_time
+    logging.info('Time to download: {}'.format(download_time.total_seconds()))
+    for video_id, entry in audio_dict.items():
+        label = label_selection(entry.labels, labels_set)
```

Comment:
Maybe `label_selection()` -> `is_positive_example()` and return a boolean THEN convert it to the actual tensorflow label below for clarity?

```
example = [1 if is_positive else 0]
```

Reply:
Done.